[docs] governance documents (#24801)
vllm-project/vllm: docs/governance/committers.md

* [@luccafong](https://github.com/luccafong): Llama models, speculative decoding, distributed
* [@mgoin](https://github.com/mgoin): Quantization and performance
* [@NickLucche](https://github.com/NickLucche): KV connector
* [@njhill](https://github.com/njhill): Distributed, API server, engine core
* [@patrickvonplaten](https://github.com/patrickvonplaten): Mistral models, new model support
* [@ProExpertProg](https://github.com/ProExpertProg): Compilation
* [@robertgshaw2-redhat](https://github.com/robertgshaw2-redhat): Core, distributed, disagg
* [@ruisearch42](https://github.com/ruisearch42): Pipeline parallelism, Ray Support
* [@russellb](https://github.com/russellb): Structured output, engine core, security 
* [@sighingnow](https://github.com/sighingnow): Qwen models, new model support
* [@tdoublep](https://github.com/tdoublep): State space models 
* [@tlrmchlsmth](https://github.com/tlrmchlsmth): Kernels and performance
* [@yaochengji](https://github.com/yaochengji): TPU integration
* [@yewentao256](https://github.com/yewentao256): Kernels and performance
* [@Yikun](https://github.com/Yikun): Pluggable hardware interface

> * [@ywang96](https://github.com/ywang96): Multi-modality, benchmarks

Reviewer: ```suggestion
* [@ywang96](https://github.com/ywang96): Multimodality, benchmarks
```

This section breaks down the active committers by vLLM components and lists the area owners.
If you have PRs touching the area, please ping the area owner for review. 

### Engine Core
* Scheduler: the core vLLM engine loop scheduling requests to next batch
    * @WoosukKwon, @robertgshaw2-redhat, @njhill, @heheda12345
* KV Cache Manager: memory management layer within scheduler maintaining KV cache logical block data
    * @heheda12345, @WoosukKwon
* AsyncLLM: the zmq based protocol hosting engine core and making it accessible for entrypoints
    * @robertgshaw2-redhat, @njhill, @russellb
* ModelRunner, Executor, Worker: the abstractions for engine wrapping model implementation
    * @WoosukKwon, @tlrmchlsmth, @heheda12345, @LucasWilkinson
* KV Connector: Connector interface and implementation for KV cache offload and transfer
    * @robertgshaw2-redhat, @njhill, @KuntaiDu, @NickLucche
* Distributed, Parallelism, Process Management: Process launchers managing each worker, and assign them to the right DP/TP/PP/EP ranks
    * @youkaichao, @njhill, @WoosukKwon, @ruisearch42
* Collectives: the usage of nccl and other communication libraries/kernels
    * @tlrmchlsmth, @youkaichao

> * Multi-modality engine and memory management: core scheduling and memory management concerning multi-modality
>     * @ywang96, @DarkLight1337, @Isotr0py

Reviewer: ```suggestion
* Multi-modality engine and memory management: core scheduling and memory management concerning multi-modality
    * @ywang96, @DarkLight1337
```

### Model Implementations
* Model Interface: The `nn.Module` interface and implementation for various models

>     * @zhuohan123, @mgoin, @simon-mo, @houseroad

Reviewer: Add either me or Cyrus for MM logics?

* Logits Processors / Sampler: The provided sampler class and pluggable logits processors
    * @njhill, @houseroad
* Custom Layers: Utility layers in vLLM such as rotary embedding and rms norms
    * @ProExpertProg
* Attention: Attention interface for paged attention
    * @WoosukKwon, @LucasWilkinson, @heheda12345
* FusedMoE: FusedMoE kernel, Modular kernel framework, EPLB
    * @tlrmchlsmth
* Quantization: Various quantization config, weight loading, and kernel.
    * @mgoin

> * MM Input Processing: Component that turns image bytes/url into feature tensors
>     * @DarkLight1337, @ywang96

Reviewer: ```suggestion
* Multi-modal Input Processing: Components that load and process image/video/audio data into feature tensors
    * @DarkLight1337, @ywang96, @Isotr0py
```

* torch compile: The torch compile integration for vLLM
    * @ProExpertProg, @zou3519, @youkaichao
* State space models: The state space models implementation in vLLM
    * @tdoublep, @tlrmchlsmth

### Entrypoints
* LLM Class: The LLM class for offline inference
    * @DarkLight1337
* API Server: The OpenAI-compatible API server
    * @DarkLight1337, @njhill, @aarnphm, @simon-mo, @heheda12345 (Responses API)
* Batch Runner: The OpenAI-compatible batch runner
    * @simon-mo

### Features
* Spec Decode: Covers model definition, attention, sampler, and scheduler related to n-grams, EAGLE, and MTP.
    * @WoosukKwon, @benchislett, @luccafong
* Structured Output: The structured output implementation
    * @russellb, @aarnphm
* RL: The RL related features such as collective rpc, sleep mode, etc.
    * @youkaichao, @zhuohan123, @22quinn
* LoRA: @jeejeelee
* Observability: Metrics and Logging
    * @robertgshaw2-redhat, @simon-mo

### Code Base
* Config: Configuration registration and parsing
    * @hmellor
* Documentation: @hmellor, @DarkLight1337, @simon-mo

> * Benchmarks: @ywang96, @DarkLight1337, @simon-mo

Reviewer: ```suggestion
* Benchmarks: @ywang96, @simon-mo
```
Maybe add @yeqcharlotte?

* CI, Build, Release Process: @khluu, @njhill, @simon-mo
* Security: @russellb

### Kernels
* FlashAttention: @LucasWilkinson
* FlashInfer: @LucasWilkinson, @mgoin, @WoosukKwon
* Blackwell Kernels: @mgoin
* DeepEP/DeepGEMM/pplx: @yewentao256
	
### Integrations

> * Hugging Face: @hmellor

Reviewer: ```suggestion
* Hugging Face: @hmellor, @Isotr0py 
```

### Engine Core

- Scheduler: the core vLLM engine loop scheduling requests to next batch
  - @WoosukKwon, @robertgshaw2-redhat, @njhill, @heheda12345
- KV Cache Manager: memory management layer within scheduler maintaining KV cache logical block data
  - @heheda12345, @WoosukKwon
- AsyncLLM: the zmq based protocol hosting engine core and making it accessible for entrypoints
  - @robertgshaw2-redhat, @njhill, @russellb
- ModelRunner, Executor, Worker: the abstractions for engine wrapping model implementation
  - @WoosukKwon, @tlrmchlsmth, @heheda12345, @LucasWilkinson
- KV Connector: Connector interface and implementation for KV cache offload and transfer
  - @robertgshaw2-redhat, @njhill, @KuntaiDu, @NickLucche
- Distributed, Parallelism, Process Management: Process launchers managing each worker, and assign them to the right DP/TP/PP/EP ranks
  - @youkaichao, @njhill, @WoosukKwon, @ruisearch42
- Collectives: the usage of nccl and other communication libraries/kernels
  - @tlrmchlsmth, @youkaichao
- Multimodality engine and memory management: core scheduling and memory management concerning vision, audio, and video inputs.
  - @ywang96, @DarkLight1337

> ### Model Implementations

Reviewer 2: I think we can also add model reasoning parser's governance in this chapter.

- Model Interface: The `nn.Module` interface and implementation for various models

>   - @zhuohan123, @mgoin, @simon-mo, @houseroad, @ywang96 (multimodality)

Reviewer 3: Add @Isotr0py (multimodality, quantization) and @jeejeelee (lora) as well

  - @WoosukKwon, @benchislett, @luccafong
- Structured Output: The structured output implementation
  - @russellb, @aarnphm
- RL: The RL related features such as collective rpc, sleep mode, etc.
  - @youkaichao, @zhuohan123, @22quinn
- LoRA: @jeejeelee
- Observability: Metrics and Logging
  - @robertgshaw2-redhat, @simon-mo

### Code Base

- Config: Configuration registration and parsing
  - @hmellor
- Documentation: @hmellor, @DarkLight1337, @simon-mo
- Benchmarks: @ywang96, @simon-mo
- CI, Build, Release Process: @khluu, @njhill, @simon-mo
- Security: @russellb

### Kernels

- FlashAttention: @LucasWilkinson
- FlashInfer: @LucasWilkinson, @mgoin, @WoosukKwon
- Blackwell Kernels: @mgoin
- DeepEP/DeepGEMM/pplx: @yewentao256

### Integrations

- Hugging Face: @hmellor, @Isotr0py
- Ray: @ruisearch42
- NIXL: @robertgshaw2-redhat, @NickLucche

### Model Definitions and Integrations

- gpt-oss: @heheda12345, @simon-mo, @zhuohan123
- Llama: @luccafong
- Qwen: @sighingnow
- Mistral: @patrickvonplaten

### Hardware

- Plugin Interface: @youkaichao, @Yikun
- AMD GPU: @gshtras
- Intel CPU/GPU: @jikunshang, @bigPYJ1151

> - Google TPU: @yaochengji

Reviewer 4: ```
- Intel Gaudi [@xuechendi](https://github.com/xuechendi) and [@kzawora-intel](https://github.com/kzawora-intel)
```
+1, here is info for Intel Gaudi plugin

    - @ProExpertProg, @zou3519, @youkaichao
- State space models: The state space models implementation in vLLM
    - @tdoublep, @tlrmchlsmth
- Reasoning and tool calling parsers
    - @chaunceyjiang, @aarnphm

### Entrypoints

- LLM Class: The LLM class for offline inference
    - @DarkLight1337
- API Server: The OpenAI-compatible API server
    - @DarkLight1337, @njhill, @aarnphm, @simon-mo, @heheda12345 (Responses API)
- Batch Runner: The OpenAI-compatible batch runner
    - @simon-mo

### Features

- Spec Decode: Covers model definition, attention, sampler, and scheduler related to n-grams, EAGLE, and MTP.
    - @WoosukKwon, @benchislett, @luccafong
- Structured Output: The structured output implementation
    - @russellb, @aarnphm
- RL: The RL related features such as collective rpc, sleep mode, etc.
    - @youkaichao, @zhuohan123, @22quinn
- LoRA: @jeejeelee
- Observability: Metrics and Logging
    - @robertgshaw2-redhat, @simon-mo

### Code Base

- Config: Configuration registration and parsing
    - @hmellor
- Documentation: @hmellor, @DarkLight1337, @simon-mo
- Benchmarks: @ywang96, @simon-mo
- CI, Build, Release Process: @khluu, @njhill, @simon-mo
- Security: @russellb

> ### Kernels

Author: We can iterate on it over time.

- Distributed, Parallelism, Process Management: Process launchers managing each worker, and assign them to the right DP/TP/PP/EP ranks
    - @youkaichao, @njhill, @WoosukKwon, @ruisearch42
- Collectives: the usage of nccl and other communication libraries/kernels
    - @tlrmchlsmth, @youkaichao
- Multimodality engine and memory management: core scheduling and memory management concerning vision, audio, and video inputs.
    - @ywang96, @DarkLight1337

### Model Implementations

- Model Interface: The `nn.Module` interface and implementation for various models
    - @zhuohan123, @mgoin, @simon-mo, @houseroad, @ywang96 (multimodality), @jeejeelee (lora)
- Logits Processors / Sampler: The provided sampler class and pluggable logits processors
    - @njhill, @houseroad, @22quinn
- Custom Layers: Utility layers in vLLM such as rotary embedding and rms norms
    - @ProExpertProg
- Attention: Attention interface for paged attention
    - @WoosukKwon, @LucasWilkinson, @heheda12345
- FusedMoE: FusedMoE kernel, Modular kernel framework, EPLB
    - @tlrmchlsmth
- Quantization: Various quantization config, weight loading, and kernel.

>     - @mgoin, @Isotr0py, @yewentao256 

Reviewer 5: ```suggestion
    - @mgoin, @Isotr0py, @yewentao256
```

    - @tdoublep, @tlrmchlsmth
- Reasoning and tool calling parsers
    - @chaunceyjiang, @aarnphm

### Entrypoints

- LLM Class: The LLM class for offline inference
    - @DarkLight1337
- API Server: The OpenAI-compatible API server
    - @DarkLight1337, @njhill, @aarnphm, @simon-mo, @heheda12345 (Responses API)
- Batch Runner: The OpenAI-compatible batch runner
    - @simon-mo

### Features

- Spec Decode: Covers model definition, attention, sampler, and scheduler related to n-grams, EAGLE, and MTP.
    - @WoosukKwon, @benchislett, @luccafong
- Structured Output: The structured output implementation
    - @russellb, @aarnphm
- RL: The RL related features such as collective rpc, sleep mode, etc.
    - @youkaichao, @zhuohan123, @22quinn
- LoRA: @jeejeelee
- Observability: Metrics and Logging
    - @robertgshaw2-redhat, @simon-mo

### Code Base

- Config: Configuration registration and parsing
    - @hmellor
- Documentation: @hmellor, @DarkLight1337, @simon-mo
- Benchmarks: @ywang96, @simon-mo
- CI, Build, Release Process: @khluu, @njhill, @simon-mo
- Security: @russellb

### Kernels

- FlashAttention: @LucasWilkinson
- FlashInfer: @LucasWilkinson, @mgoin, @WoosukKwon

> - Blackwell Kernels: @mgoin
> - DeepEP/DeepGEMM/pplx: @yewentao256

Reviewer 5: ```suggestion
- Blackwell Kernels: @mgoin, @yewentao256 
- DeepEP/DeepGEMM/pplx: @mgoin, @yewentao256
```
